tests/checkpointing/test_model_checkpoint.py

```diff
@@ -967,15 +966,13 @@ def assert_checkpoint_log_dir(idx):
         assert_checkpoint_content(ckpt_dir)
 
         # load from checkpoint
-        trainer_config["callbacks"] = [ModelCheckpoint(dirpath=ckpt_dir, save_top_k=-1)]
```

Comment:
Need to remove this because creating a new instance means we lose the existing value for
`ModelCheckpoint._last_global_step_saved`

which avoids saving a duplicate checkpoint on the second fit call. Perhaps this value should be saved and reloaded so that creating a new instance also works.